pkg/util/azureclient/resources.go

```diff
@@ -0,0 +1,35 @@
+package azureclient
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// DeploymentsClient is a minimal interface for azure DeploymentsClient
+type DeploymentsClient interface {
+	CreateOrUpdate(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) (result resources.DeploymentsCreateOrUpdateFuture, err error)
+	Client
```

Comment:
The interface for this client (and probably some of the rest too) is not that intuitive. Here we either need to CreateOrUpdate a deployment and we need to WaitForCompletion. I feel it would be better to expose WaitForCompletion in the interface as opposed to Client.

Reply:
Same comment applies for the `VirtualMachineScaleSetVMsClient` and `VirtualMachineScaleSetsClient` interfaces.

Reply:
No blockers but please fix in a follow-up

/lgtm